clone-object.js

```diff
@@ -0,0 +1,15 @@
+describe('clone object', function () {
+  it('should clone an object', function () {
+    var expected = {name: 'Ahmed', age: 27, skills: ['cycling', 'walking', 'eating']},
+        obj = {};
+		if (expected instanceof Object){
+		obj = expected.constructor();
+		for (var attr in expected) {
+			if (expected.hasOwnProperty(attr)) 
+				obj[attr] = expected[attr];

```

Comment:
No, it won't be by reference. As it has to copy the object properties to the other.